File: 3451-string-compression-iii/3451-string-compression-iii.cs
public class Solution {
    public string CompressedString(string word) {
        StringBuilder comp = new StringBuilder(word.Length);
        int cnt = 0;
        char prev = word[0];
        foreach (var c in word) {
            if (prev == c) {
                cnt++;
                if (cnt == 9) {
                    comp.Append(cnt);
                    comp.Append(prev);
                    cnt = 0;
                }
            }
            else{
                if (cnt > 0) {
                    comp.Append(cnt);
                    comp.Append(prev);
                }
                cnt = 1;
                prev = c;
            }
        }
        if (cnt > 0) {
            comp.Append(cnt);
            comp.Append(prev);
        }
        return comp.ToString();
    }
}

File: 3451-string-compression-iii/3451-string-compression-iii.cpp
class Solution {
public:
    string compressedString(string word) {
        string comp = "";
        int i = 0;
        while(i < word.length()){
            int cnt = 1;
            char curr_char = word.at(i);
            while((i + 1) < word.length() and word[i + 1] == curr_char and cnt < 9){
                cnt += 1;
                i++;
            }
            comp += to_string(cnt) + curr_char;
            i++;
        }
        return comp;
    }
};